ttnn/cpp/ttnn/operations/data_movement/pad/device/pad_program_factory.cpp

```diff
@@ -1575,4 +1577,143 @@ operation::ProgramWithCallbacks pad_rm_sharded(
     return {.program = std::move(program), .override_runtime_arguments_callback = override_runtime_args_callback};
 }
 
+operation::ProgramWithCallbacks pad_rm_sharded_width_only(
+    const Tensor& input_tensor,
+    Tensor& output,
+    const tt::tt_metal::LegacyShape& output_tensor_shape,
+    const ttnn::SimpleShape& input_tensor_start,
+    float pad_value) {
+    Program program{};
+
+    TT_ASSERT(
+        output.shard_spec().has_value() and output.shard_spec()->shape[1] == output_tensor_shape[-1],
+        "output must be sharded with shard width equal to output width");
+
+    uint32_t W = input_tensor.shape()[3];
+    uint32_t W_padded = output_tensor_shape[3];
+
+    auto unpadded_stick_bytes = W * input_tensor.element_size();
+    auto padded_stick_bytes = W_padded * input_tensor.element_size();
+
+    TT_FATAL(unpadded_stick_bytes != padded_stick_bytes,
+             "ttnn.pad: pad_rm_sharded_width_only should only be used with non-zero width padding.");
```

Comment:
I read this error and I am not sure, what do I change as a user?

Reply:
I actually decided to just remove this, since it's a silly thing to do but not an error per-se.